api/include/opentelemetry/std/span.h

```diff
@@ -11,7 +11,7 @@
 #if defined __has_include
 #  if __has_include(<version>)  // Check for __cpp_{feature}
 #    include <version>
-#    if defined(__cpp_lib_span)
+#    if defined(__cpp_lib_span) && __cplusplus > 201703L
```

Comment:
because  `__cpp_lib_span` is defined for C++17 on g++-10:`g++-10 (Ubuntu 10.3.0-1ubuntu1~20.04) 10.3.0`.
I use these CMake options for build:
`-DCMAKE_CXX_COMPILER=g++-10 -DCMAKE_CXX_STANDARD=17`

Reply:
Which should be ok right? If the `HAVE_CPP_STDLIB` option is specified, and the standard library provides the implementation ( stable or experimental) it should get selected. What is the behavior now with these changes on `g++-10/C++17` - compilation failure as std::span is not available?

Reply:
The behavior for `g++-10/C++17` is that `nostd::span` will be picked if the `WITH_GSL` option was not used. I think we shouldn't use `std::span` if the user explicitly asks for `C++17`.
Shall I enable `std::span` if `C++17` was used and `std::span` is available?